packages/service-library/src/servicelib/volumes_utils.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Final, Optional
+
+import aiofiles
+from pydantic import BaseModel, Field, root_validator
+
+HIDDEN_FILE_NAME: Final[str] = ".hidden_do_not_remove"
+AGENT_FILE_NAME: Final[str] = ".agent"
+
+
+class VolumeState(BaseModel):
+    requires_saving: bool = Field(
+        ...,
+        description=(
+            "when True, the contents of the volume must be saved whe "
+            "closing the service"
+        ),
+    )
+    was_saved: Optional[bool] = Field(
+        None,
+        description=(
+            "When `True`: the sidecar managed to save the contents of the volume. "
+            "When `False`: there was an issue and the contes was not saved, it is "
+            "up to the agent to save the content of the volume. "
+            "When `None`: if requires_saving is False, it makes more sense "
+            "to have this se to None since it will not be used"
+        ),
+    )
+
+    @root_validator
+    @classmethod
+    def enforce_use_cases(cls, values: dict) -> dict:
+        requires_saving: bool = values["requires_saving"]
+        was_saved: Optional[bool] = values["was_saved"]
+
+        # requires_saving is False was_saved MUST be None
+        if requires_saving is False and was_saved is not None:
+            raise ValueError(f"When {requires_saving=}, was_saved must be None")
```

Comment:
so for instance I do not understand the usefulness of such a validation. if requires_saving is False, then we do not care about was_saved right? is there any other use case I'm missing here?